mmrotate/datasets/dior.py

```diff
@@ -0,0 +1,216 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import os.path as osp
+import xml.etree.ElementTree as ET
+from typing import List, Optional, Union
+
+import mmcv
+import numpy as np
+from mmengine.dataset import BaseDataset
+from mmengine.fileio import FileClient, list_from_file
+
+from mmrotate.registry import DATASETS
+
+
+@DATASETS.register_module()
+class DIORDataset(BaseDataset):
+    """DIOR dataset for detection.
+
+    Args:
+        ann_subdir (str): Subdir where annotations are.
+            Defaults to 'Annotations/Oriented Bounding Boxes/'.
+        file_client_args (dict): Arguments to instantiate a FileClient.
+            See :class:`mmengine.fileio.FileClient` for details.
+            Defaults to ``dict(backend='disk')``.
+        xmltype (str): Choose obb or hbb as ground truth.
+            Defaults to `obb`.
+    """
+
+    METAINFO = {
+        'CLASSES':
+        ('airplane', 'airport', 'baseballfield', 'basketballcourt', 'bridge',
+         'chimney', 'expressway-service-area', 'expressway-toll-station',
+         'dam', 'golffield', 'groundtrackfield', 'harbor', 'overpass', 'ship',
+         'stadium', 'storagetank', 'tenniscourt', 'trainstation', 'vehicle',
+         'windmill'),
+        # PALETTE is a list of color tuples, which is used for visualization.
+        'PALETTE': [(220, 20, 60), (119, 11, 32), (0, 0, 142), (0, 0, 230),
+                    (106, 0, 228), (0, 60, 100), (0, 80, 100), (0, 0, 70),
+                    (0, 0, 192), (250, 170, 30), (100, 170, 30), (220, 220, 0),
+                    (175, 116, 175), (250, 0, 30), (165, 42, 42),
+                    (255, 77, 255), (0, 226, 252), (182, 182, 255), (0, 82, 0),
+                    (120, 166, 157)]
+    }
+
+    def __init__(self,
+                 ann_subdir: str = 'Annotations/Oriented Bounding Boxes/',
+                 file_client_args: dict = dict(backend='disk'),
+                 xmltype: str = 'obb',
```

Comment:
`xmltype` may not well understood, maybe `ann_type` or other name?

Reply:
Thanks for your advice. I have updated.